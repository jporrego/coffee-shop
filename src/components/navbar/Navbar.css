.navbar {
    position: sticky;
    display: flex;
    justify-content: center;
    top: 0;
    width: 100%;
    height: 7.5rem;
    padding: 0rem 2rem;
    background-color: var(--white);
    z-index: 100;
    box-shadow: 0px 5px 5px rgba(0, 0, 0, 0.1);
}

.navbar-content {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 100%;
}

.navbar a {
    font-size: 2.2rem;
    font-weight: 700;
    text-transform: capitalize;
    color: black;
    text-decoration: none;
    transition: all 0.15s ease-out;
}

.navbar a:hover {
    color: var(--brown-dark);
}

h1 {
    font-size: 2.2rem;
}

@media only screen and (min-width: 1024px) {
    .navbar-content {
        width: 80%;
    }
}