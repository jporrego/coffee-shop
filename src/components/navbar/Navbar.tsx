import React from "react";
import "./Navbar.css";
import { StringTouple } from "../../types";

interface NavbarProps {
  paths?: {
    path: string;
    text: string;
  }[];
  cart?: React.ReactNode;
}
const Navbar: React.FC<NavbarProps> = ({ paths, cart }) => {
  const renderLinks = () => {
    if (paths !== undefined) {
      return; /*paths.map((link) => (
        <Link key={link.path} to={link.path}>
          {link.text}
        </Link>
      ));*/
    }
  };
  return (
    <div className="navbar">
      <div className="navbar-content">
        <h1>Shop</h1>
        {cart && cart}
      </div>
    </div>
  );
};

export default Navbar;
