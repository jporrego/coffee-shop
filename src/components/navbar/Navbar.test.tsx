import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Navbar from "./Navbar";

describe("navbar", () => {
  it("renders correctly", () => {
    // since screen does not have the container property, we'll destructure render to obtain container for this test
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });
});
