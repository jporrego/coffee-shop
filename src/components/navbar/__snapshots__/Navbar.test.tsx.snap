// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders correctly 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navbar-content"
    >
      <h1>
        CaféShop
      </h1>
      <div
        class="navbar-links"
      >
        <a
          class="link"
          href="#"
        >
          Home
        </a>
        <a
          class="link current"
          href="#"
        >
          Shop
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`navbar renders the cart component if it's passed as a prop 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navbar-content"
    >
      <h1>
        CaféShop
      </h1>
      <div
        class="navbar-links"
      >
        <a
          class="link"
          href="#"
        >
          Home
        </a>
        <a
          class="link current"
          href="#"
        >
          Shop
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`product renders the product 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="navbar-content"
    >
      <h1>
        CaféShop
      </h1>
      <div
        class="navbar-links"
      >
        <a
          class="link"
          href="#"
        >
          Home
        </a>
        <a
          class="link current"
          href="#"
        >
          Shop
        </a>
      </div>
    </div>
  </div>
</div>
`;
